# IS 362 – Week 5 Assignment

In [2]:
import pandas as pd

airports = pd.read_csv('C:\\Users\Victor Pessoa\\Desktop\\IS 362 – Week 5 Assignment\\airports.csv')
weather = pd.read_csv('C:\\Users\\Victor Pessoa\\Desktop\\IS 362 – Week 5 Assignment\\weather.csv')


## Northernmost Airport

In [4]:
northernmost_airports = airports.sort_values(by='lat', ascending=False).head(5)

print("The top 5 northernmost airports in the United States are:")
print(northernmost_airports[['name', 'lat']])

The top 5 northernmost airports in the United States are:
                                           name        lat
417                     Dillant Hopkins Airport  72.270833
230                  Wiley Post Will Rogers Mem  71.285446
110                          Wainwright Airport  70.638056
708                               Wainwright As  70.613378
152  Atqasuk Edward Burnell Sr Memorial Airport  70.467300


### Given that according to google maps the corrected latitude for Dillant Hopkins Airport is approximately 42.9142.91 and the longitude is approximately −72.27, this suggests that the original latitude value in the dataset is likely incorrect and should be validated.

In [7]:
airports.loc[airports['name'] == 'Dillant Hopkins Airport', 'lat'] = 42.90652313880002

northernmost_airports_corrected = airports.sort_values(by='lat', ascending=False).head(5)

print("The top 5 northernmost airports in the United States are (corrected):")
print(northernmost_airports_corrected[['name', 'lat']])


The top 5 northernmost airports in the United States are (corrected):
                                            name        lat
230                   Wiley Post Will Rogers Mem  71.285446
110                           Wainwright Airport  70.638056
708                                Wainwright As  70.613378
152   Atqasuk Edward Burnell Sr Memorial Airport  70.467300
1363                        Ugnu-Kuparuk Airport  70.330833


## Easternmost Airport

In [5]:
easternmost_airports = airports.sort_values(by='lon', ascending=False).head(5)

print("The top 5 easternmost airports in the United States are:")
print(easternmost_airports[['name', 'lon']])

The top 5 easternmost airports in the United States are:
                               name         lon
1290                   Eareckson As  174.113620
942                Montgomery Field  117.759000
396   Deer Valley Municipal Airport  112.457000
417         Dillant Hopkins Airport   42.898333
444      Eastport Municipal Airport  -67.012694


### According to Google Maps I got:
##### 54.21888484099894, 174.06567797595457 for Eareckson As,
##### 32.826042630263515, -117.1016092873497 for Montgomery Field,
##### 33.97864672701835, -118.42136444041208 for Deer Valley Municipal Airport,
##### 42.90664207509538, -72.27254825923286 for Dillant Hopkins Airport,
##### and 45.071980546937176, -66.91126034461944 for Eastport Municipal Airport

In [13]:
airports.loc[airports['name'] == 'Montgomery Field', 'lon'] = -117.1016092873497
airports.loc[airports['name'] == 'Deer Valley Municipal Airport', 'lon'] = -118.42136444041208
airports.loc[airports['name'] == 'Dillant Hopkins Airport', 'lon'] = -72.27254825923286

easternmost_airport_corrected = airports.sort_values(by='lon', ascending=False).head(5)

print("The top 5 easternmost airport in the United States are (corrected):")
print(easternmost_airport_corrected[['name', 'lon']])

The top 5 easternmost airport in the United States are (corrected):
                                     name         lon
1290                         Eareckson As  174.113620
444            Eastport Municipal Airport  -67.012694
624                          Houlton Intl  -67.792056
259                          Caribou Muni  -68.017917
1101  Northern Maine Rgnl At Presque Isle  -68.044797


## Windiest Weather on February 12th, 2013

In [12]:
windiest_weather = weather[
    (weather['year'] == 2013) & 
    (weather['month'] == 2) & 
    (weather['day'] == 12)
].sort_values(by='wind_speed', ascending=False).head(3)

print("The New York area airport with the windiest weather on February 12th, 2013 is:")
print(windiest_weather[['origin', 'wind_speed']])


The New York area airport with the windiest weather on February 12th, 2013 is:
      origin  wind_speed
1009     EWR  1048.36058
18417    LGA    23.01560
1018     EWR    21.86482


### A wind speed of 1048.36058 seems extremely high and is likely an error in the data. The actual The New York area airport with the windiest weather on February 12th, 2013 would be LGA with wind speed of 23.01560.

### Conclusion

In this analysis, we aimed to identify the northernmost, easternmost airports in the United States and the New York area airport with the windiest weather on February 12th, 2013.

- **Northernmost Airport:** The initial analysis identified Dillant Hopkins Airport as the northernmost airport, but this was found to be incorrect. After validating and correcting the data, Wiley Post Will Rogers Mem emerged as the northernmost airport with a latitude of 71.285446.

- **Easternmost Airport:** The initial analysis had inaccuracies due to positive longitude values. After corrections, Eareckson As emerged as the easternmost, considering U.S. territories, with a longitude of 174.06567797595457. However, within the contiguous United States, Eastport Municipal Airport is the easternmost.

- **Windiest Weather on February 12th, 2013:** The initial analysis revealed an outlier with an extremely high wind speed at EWR. After careful consideration, this value was deemed incorrect, and LGA was identified as having the windiest weather with a wind speed of 23.01560.

The analysis underscores the critical importance of data validation and the handling of outliers in deriving accurate insights from any data-driven analysis. The corrected data provided more plausible results, aligned with external validations, leading to more reliable conclusions.

